### DISCUSSION TITLE
How can I integrate Goatcounter in SvelteKit SSR with hydration? "window.addEventListener('hashchange'," seems not to work

### DISCUSSION THREAD

**Discussion by stephane-klein** [sveltejs/kit#10303] [Q&A]
Hello,

I try to configure [Add GoatCounter to a SPA](http://127.0.0.1:8080/help/spa) in my SvelteKit project, [here](https://github.com/stephane-klein/sklein.xyz/blob/4284e61399db4fd31cb02ece7a40ec4a5b00b8d3/src/app.html#L16).

```
            window.addEventListener('hashchange', function(e) {
                console.log('ici1');
                window.goatcounter.count({
                    path: location.pathname + location.search + location.hash,
                })
            })
```

I don't see `"ici1"` in my browser console when I change pages :thinking:.

Do you have any ideas for implementing this [Goatcounter](https://www.goatcounter.com/) integration in my SvelteKit project?

Best regards,  
Stéphane

**Comment by geoffrich** [comment]
Listening to the `hashchange` event won't work because SvelteKit does not use hash-based navigation.

I haven't tried it myself, but maybe you can use `afterNavigate` in your root `+layout.svelte`?

```html
<script>
	import { afterNavigate } from '$app/navigation';

	afterNavigate(({ to }) => {
		const url = to?.url;
		if (url) {
			window.goatcounter.count({
				path: url.pathname + url.search + url.hash
			})
		}
	})
</script>

<slot />
```
